pytorch_pfn_extras/runtime/_runtime.py

```diff
@@ -73,14 +74,16 @@ def convert_batch(self, args: Any) -> Any:
             # namedtuple
             return args._replace(  # type: ignore[attr-defined]
                 **self._convert_batch_dict(args._asdict()))  # type: ignore
-        if isinstance(args, dict):
+        elif hasattr(args, "__dataclass_fields__"):
+            return self._convert_batch_dict(dataclasses.asdict(args))
+        elif isinstance(args, dict):
             return self._convert_batch_dict(args)
-        if isinstance(args, (list, tuple)):
+        elif isinstance(args, (list, tuple)):
             return [
                 self.move_tensor(v) if isinstance(v, torch.Tensor) else v
                 for v in args
             ]
-        if isinstance(args, torch.Tensor):
+        elif isinstance(args, torch.Tensor):
             return self.move_tensor(args)
         return args
```

Comment:
For the consistency of fixes.
```suggestion
        else:
            return args
```